aggregator_core/src/datastore.rs

```diff
@@ -1491,21 +1519,59 @@ impl<C: Clock> Transaction<'_, C> {
                       AND client_reports.client_timestamp < UPPER($2::TSRANGE)
                       AND client_reports.client_timestamp >= $3
                       AND client_reports.aggregation_started = FALSE
-                ) AS unaggregated_report_exists",
+                ) AS never_aggregated_report_exists",
             )
             .await?;
-        let row = self
+        let never_aggregated_report_exists = self
             .query_one(
-                &stmt,
+                &fast_path_stmt,
                 &[
                     /* task_id */ &task_info.pkey,
                     /* batch_interval */ &SqlInterval::from(batch_interval),
                     /* threshold */
                     &task_info.report_expiry_threshold(&self.clock.now().as_naive_date_time()?)?,
                 ],
             )
+            .await?
+            .get("never_aggregated_report_exists");
+
+        // TODO(timg): are we OK with runtime type check?
+        if never_aggregated_report_exists || TypeId::of::<AP>() == TypeId::of::<()>() {
+            return Ok(never_aggregated_report_exists);
+        }
```

Comment:
I want to draw reviewer attention here: we've previously used run-time checks via the [`Any` trait](https://doc.rust-lang.org/std/any/index.html) to see if an aggregation parameter is `()`, but only in test code (see tests in `aggregation_job_creator.rs`). The alternative to this would be provide distinct versions of `interval_has_unaggregated_reports` for the trivial agg param case and the non-trivial case, and dispatch appropriately, probably all the way up in [`CollectionJobDriver::step_collection_job`](https://github.com/divviup/janus/blob/ee60cba2cb1144f9a77052e669f6a0ec32142e6d/aggregator/src/aggregator/collection_job_driver.rs#L91).